nixos/modules/services/security/bitwarden_rs/default.nix

```diff
@@ -0,0 +1,177 @@
+        Group under which ${description} runs.
+      '';
+    };
+
+    dataFolder = mkOption {
+      type = str;
+      default = "/var/lib/bitwarden_rs";
+      description = ''
+        The folder under which ${description} will preserve any persistent data.
+      '';
+    };
+
+    enableWebVault = mkEnableOption "Bitwarden Web Vault" // {
+      default = true;
+    };
+
+    enableWebsocket = mkEnableOption "websocket notifications";
+
+    backupFolder = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        The folder under which ${description} will backup its persistent data.
+      '';
+    };
+
+    backupUser = mkOption {
+      type = str;
+      default = "bitwarden_rs";
+      description = ''
+        User under which ${description} backups are run.
+      '';
+    };
+
+    backupGroup = mkOption {
+      type = str;
+      default = "bitwarden_rs";
+      description = ''
+        Group under which ${description} backups are run.
+      '';
+    };
+
+    backupOn = mkOption {
+      type = str;
+      default = "23:00";
+      description = ''
+        On what time ${description} backups are run.
+      '';
+    };
+
+    config = mkOption {
+      type = attrsOf (nullOr (either (either bool int) str));
+      default = {};
+      example = literalExample ''
+        {
+          signupsAllowed = true;
+          domain = https://example.com;
+          rocketPort = 8222;
+          rocketLog = "critical";
+        }
+      '';
+      description = ''
+        Config environment variables passed to the bitwarden.env file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.bitwarden_rs.config.webVaultFolder = mkIf cfg.enableWebVault (mkDefault "${pkgs.bitwarden_rs-vault}/share/bitwarden_rs/vault");
+
+    users.extraUsers = optional (cfg.user == "bitwarden_rs") {
+      name = "bitwarden_rs";
+      inherit description;
+      group = "bitwarden_rs";
+    };
+
+    users.extraGroups = optional (cfg.group == "bitwarden_rs") {
+      name = "bitwarden_rs";
+    };
+
+    environment.systemPackages = [ pkgs.bitwarden_rs ];
+
+    systemd.services.bitwarden_rs = {
+      inherit description;
+      after = [ "network.target" ];
+      path = with pkgs; [ openssl ];
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        EnvironmentFile = configFile;
+        ExecStart = "${pkgs.bitwarden_rs}/bin/bitwarden_rs";
+        LimitNOFILE = "1048576";
+        LimitNPROC = "64";
+        PrivateTmp = "true";
+        PrivateDevices = "true";
+        ProtectHome = "true";
+        ProtectSystem = "strict";
+        WorkingDirectory = cfg.dataFolder;
+        ReadWriteDirectories = cfg.dataFolder;
+        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
+        StateDirectory = mkIf (cfg.dataFolder == "/var/lib/bitwarden_rs") "bitwarden_rs";
+      };
+      wantedBy = [ "multi-user.target" ];
+      restartTriggers = [ configFile ];
```

Comment:
This isn't needed, because `configFile` is already part of the unit definition via `EnvironmentFile`. `restartTriggers` is only needed when changing to its arguments normally wouldn't change the unit file.